### PR TITLE
feat: Add SageMaker Embeddings

### PR DESCRIPTION
**Pull request by kbreese-x**
## Description

This pull request adds SageMaker as an embeddings model provider. The new provider is primarily designed to work with SageMaker endpoints deployed using the Huggingface TEI containers, but can be customized to work with other endpoints too through the use of `requestOptions.extraBodyProperties`. Documentation on how to set up the new provider has been added to both the Embedding Models and Sagemaker Provider pages. The config_schema has been updated for both vscode and intellij to include `sagemaker` as a new provider, as well as providing some default region options for the most commonly used AWS regions.

## Checklist

- [x] The base branch of this PR is `dev`, rather than `main`
- [x] The relevant docs, if any, have been updated or created

## Screenshots

[ For visual changes, include screenshots. ]

## Testing

Requires: AWS Sagemaker endpoint deployed with TEI container (example [here](https://www.philschmid.de/sagemaker-train-deploy-embedding-models#4-deploy--test-fine-tuned-embedding-model-on-amazon-sagemaker))
Requires: an AWS profile called "sagemaker" or a default profile with permissions to invoke endpoints

Add the following configuration:
"embeddingsProvider": {
    "provider": "sagemaker",
    "model": "embedding-test-endpoint",
    "maxBatchSize": 12,
    "profile": "default",
    "region": "us-west-2",
    "requestOptions": {
      // this is an example for an endpoint deployed using TEI
      "extraBodyProperties": {
        "parameters": {
          "truncate": true
        }
      }
    }

To learn more about the parameters you can send to TEI backed endpoints, see the API reference [here](https://huggingface.github.io/text-embeddings-inference/#/Text%20Embeddings%20Inference/embed)

### CODE REVIEW

File: docs/docs/customize/model-types/embeddings.md [hunk 1]
```diff
@@ -69,3 +69,7 @@ See [here](../model-providers/more/cohere.md#embeddings-model) for instructions
 ### Gemini
 
 See [here](../model-providers/top-level/gemini.md#embeddings-model) for instructions on how to use Gemini for embeddings.
+
```
Comment (sestinj) [inline docs/docs/customize/model-types/embeddings.md]: I think SageMaker is a bit more niche so I would prefer it not exist in this list here (we also leave out a number of other embeddings providers)

File: extensions/vscode/config_schema.json [hunk 2]
```diff
@@ -246,7 +246,24 @@
         "region": {
           "title": "Region",
           "description": "The region where the model is hosted",
-          "type": "string"
+          "anyOf":[
+            {
+              "enum": [
```
Comment (sestinj) [inline extensions/vscode/config_schema.json:251]: Can this be made a variable?

**Review by sestinj — CHANGES_REQUESTED** [review]
minor comments then good to go—thank you for this!

**Comment by kbreese-x** [comment]
Done!
didn't move the region definition to top level since it seems minor, but just referenced it where needed

File: extensions/vscode/config_schema.json (hunk 2 above)
Comment (kbreese-x) [reply extensions/vscode/config_schema.json:251]: referenced later instead of duplicating

**Review by sestinj — APPROVED** [review]

**Comment by sestinj** [comment]
Hi @kbreese-x, yesterday we shared [some updates](https://discordapp.com/channels/1108621136150929458/1131313996750917835/1371996789967687783) with our contributors about how we're aiming to improve the contribution process. Part of this included the addition of a [Contributor License Agreement](https://github.com/continuedev/continue/blob/main/docs/docs/CLA.md) (CLA) to protect both contributors and the project. We're reaching out to ask that previous contributors sign it.

Could you please take a moment to sign, or if you have any questions send me a message? (either here or nate@continue.dev would work)

To do so, you just need to **post a comment below with the following text:**<br/>
```
I have read the CLA Document and I hereby sign the CLA
```

❤️ Thank you for the work you've done on Continue, and let me know if you have any suggestions on how we can make the project even better!

**Pull request merged**